### PR TITLE
quality_factor_of_empty_rectangular_resonator

### PR DESCRIPTION
**Pull request by Zulusssss**
Add law and tests.

### CODE REVIEW

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+## In the case when the resonator is empty, its quality factor depends only on the losses in the metal walls of the resonator.
+## The quality factor shows the ratio of the energy stored in the resonator to the energy loss during one oscillation period.
+
+## Law is: Q = w0 * mu0 * mur * b * a * l * (a^2 + l^2) / (2 * Rs * (a^3 * (l + 2 * b) + l^3 * (a + 2 * b))), where
+## Q - quality factor of the resonator,
+## mur - relative permeability of the resonator walls,
+## Rs1 - surface resistance of the resonator walls,
+## a - height of the resonator,
+## b - width of the resonator,
+## l - length of the resonator.
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:18]: `w0` and `mu0` are not described.

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output, convert_to, dimensionless, angle_type)
+
+
+# Description
+## A rectangular resonator consists of metal walls and a material filling it.
+## In the case when the resonator is empty, its quality factor depends only on the losses in the metal walls of the resonator.
+## The quality factor shows the ratio of the energy stored in the resonator to the energy loss during one oscillation period.
+
+## Law is: Q = w0 * mu0 * mur * b * a * l * (a^2 + l^2) / (2 * Rs * (a^3 * (l + 2 * b) + l^3 * (a + 2 * b))), where
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:12]: Link to the law please.

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+## a - height of the resonator,
+## b - width of the resonator,
+## l - length of the resonator.
+
+# Conditions:
+# - for transverse electric waves with indices m, n, p = 1, 0, 1.
+
+quality_factor = Symbol("quality_factor", dimensionless)
+
+resonant_frequency = Symbol("resonant_frequency", angle_type / units.time)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:25]: angular?

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# Description
+## A rectangular resonator consists of metal walls and a material filling it.
+## In the case when the resonator is empty, its quality factor depends only on the losses in the metal walls of the resonator.
+## The quality factor shows the ratio of the energy stored in the resonator to the energy loss during one oscillation period.
+
+## Law is: Q = w0 * mu0 * mur * b * a * l * (a^2 + l^2) / (2 * Rs * (a^3 * (l + 2 * b) + l^3 * (a + 2 * b))), where
+## Q - quality factor of the resonator,
+## mur - relative permeability of the resonator walls,
+## Rs1 - surface resistance of the resonator walls,
+## a - height of the resonator,
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:16]: I think it should be `width`.

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py (hunk 2 above)
Comment (Zulusssss) [reply symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:12]: > Link to the law please.

https://old.study.urfu.ru/view/aid/67/1/resonators.pdf

Page 24.

File: symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py [hunk 5]
```diff
@@ -11,43 +11,45 @@
...
 ## mur - relative permeability of the resonator walls,
 ## Rs1 - surface resistance of the resonator walls,
-## a - height of the resonator,
-## b - width of the resonator,
+## a - width of the resonator,
+## b - height of the resonator,
 ## l - length of the resonator.
 
 # Conditions:
 # - for transverse electric waves with indices m, n, p = 1, 0, 1.
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/quality_factor_of_empty_rectangular_resonator.py:23]: Add to file name wave type and mode.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**